reference-implementation/lib/parser.js

```diff
@@ -122,3 +128,17 @@ function normalizeAddress(address, baseURL) {
 function isJSONObject(value) {
   return typeof value === 'object' && value !== null && !Array.isArray(value);
 }
+
+function longerLengthThenCodeUnitOrder(a, b) {
+  return compare(b.length, a.length) || compare(a, b);
```

Comment:
Super edge case, but in the fallthrough case, `a` being `"##"` and `b` being `"00"` seems like it results in non-alphanumeric ordering since "00" > 1 :P

Reply:
Same issue as typo above then.